### PR TITLE
Removing the single instance of NullValue 

### PR DESCRIPTION
**Pull request by simeonleatherland**
NullValue represent a certain Ast element, therefore it should not be used as a singleton.

### CODE REVIEW

File: src/test/groovy/graphql/parser/ParserTest.groovy [hunk 1]
```diff
@@ -516,17 +516,18 @@ class ParserTest extends Specification {
...
+        def input = "{ foo(bar: null, bell : null) }"
 
         when:
         def document = new Parser().parseDocument(input)
         def operation = document.definitions[0] as OperationDefinition
         def selection = operation.selectionSet.selections[0] as Field
 
         then:
-        selection.arguments[0].value == NullValue.Null
+        assert selection.arguments[0].value instanceof NullValue
```
Comment (andimarek) [inline src/test/groovy/graphql/parser/ParserTest.groovy]: please make sure that source location is also set

Comment (simeonleatherland) [reply src/test/groovy/graphql/parser/ParserTest.groovy]: The constant has been deleted - see changes to NullValue file. 

What do you mean by source location is set?

File: src/main/java/graphql/parser/GraphqlAntlrToLanguage.java [hunk 2]
```diff
@@ -716,7 +716,7 @@ protected Value createValue(GraphqlParser.ValueContext ctx) {
             addCommonData(booleanValue, ctx);
             return booleanValue.build();
         } else if (ctx.NullValue() != null) {
-            return Null;
+            return NullValue.newNullValue().build();
```
Comment (andimarek) [inline src/main/java/graphql/parser/GraphqlAntlrToLanguage.java]: I don't thing this is correct: we need to record the source location, ?

File: src/test/groovy/graphql/parser/ParserTest.groovy [hunk 3]
```diff
@@ -516,17 +516,21 @@ class ParserTest extends Specification {
...
+        assert selection.arguments[1].value instanceof NullValue
+
+        assert selection.arguments[0].sourceLocation.toString() == "SourceLocation{line=1, column=7}"
```
Comment (andimarek) [inline src/test/groovy/graphql/parser/ParserTest.groovy]: don't use assert: spock just takes a == b
also: this is the right test, yes, but not sure how this works, see the other comment above

Comment (bbakerman) [reply src/test/groovy/graphql/parser/ParserTest.groovy]: Spock will fail a test when any statement in the `then:` block is non true

So `selection.arguments[0].value instanceof NullValue` is a valid statement - no need for assert

Assert is needed if you create a common method that you call in a test eg

```

when:
     someCommonMethos(value)

def someCommonMethod(value) {
    assert value != null
    assert value instanceof Foo
    return true
}
```

**Pull request merged**